BookStore.h

```diff
@@ -0,0 +1,135 @@
+#ifndef UNTITLED_BOOKSTORE_H
+#define UNTITLED_BOOKSTORE_H
+#ifndef UNTITLED_H
+#define UNTITLED_H
+
+#include "string"
+#include "vector"
+#include "algorithm"
+#include "iostream"
+
+class Book {
+public:
+    Book(std::string name,
+         int price_,
+         int numberOfPages_,
+         std::string author_,
+         int quantity_,
+         int numberOfSales_);
+    ~Book();
+
+    std::string getName(){
+        return name;
+    }
+    int getPrice(){
+        return price;
+    }
+    int getNumberOfSales(){
+        return numberOfSales;
+    }
+    std::string getAuthor(){
+        return author;
+    }
+
+private:
+    std::string name;
+    int price;
+    int numberOfPages;
+    std::string author;
+    int quantity;
+    int numberOfSales;
+};
+
+class BookStore {
+public:
+    BookStore(std::vector<Book> books);
+    BookStore();
+    ~BookStore();
+
+    void addBook(Book book){
+        books.push_back(book);
+    }
+
+    void deleteBook(std::string name) {
+        int ind = 0;
+        for (int i = 0; i < books.size(); ++i) {
+            if (books[i].getName() == name) {
```

Comment:
Instead of looping through the entire vector of books to find the book with the given name, you can use the std::find_if algorithm which takes a lambda function as its argument. This lambda function should return true if the book name matches the given name and false otherwise.

Reply:
it seems to me that my method is not bad either